core/trino-main/src/main/java/io/trino/metadata/MetadataManager.java

```diff
@@ -510,11 +510,19 @@ public List<QualifiedObjectName> listTables(Session session, QualifiedTablePrefi
 
         Optional<QualifiedObjectName> objectName = prefix.asQualifiedObjectName();
         if (objectName.isPresent()) {
-            Optional<RelationType> relationType = getRelationTypeIfExists(session, objectName.get());
-            if (relationType.isPresent()) {
-                return ImmutableList.of(objectName.get());
+            try {
+                Optional<RelationType> relationType = getRelationTypeIfExists(session, objectName.get());
+                if (relationType.isPresent()) {
+                    return ImmutableList.of(objectName.get());
+                }
+                // TODO we can probably return empty list here
```

Comment:
good idea, let's do this

(pre-existing, so separate commit)

Reply:
This comment is preexisting only indention changed

Reply:
yes, i know, but still can be addressed